### PR TITLE
Improved plot options in Climatic > PICSA > Rainfall Graphs dialog

### LINKED ISSUE #7498: Small change in PICSA rainfall graphs dialogue
@N-thony has made a change in the Plot options dialogue for the Mapping dialogue.  I suggest we apply this feature to other plotting dialogues, and start with the Climatic > PICSA > Rainfall Graphs.  

Now this is different to our other graphics dialogues in a small way.  In most of the others, I mainly use the Plot Options button and then (rarely) the Box Options or Jitter Options, etc.  Here I almost always use the PICSA options button, and only rarely the more general Plot Options.
So, I suggest either:
a) Keep the PICSA Options as it is. So the 2 buttons remain.  Also put the Down arrow on the Plot Options, and include Plot Options Line Options, Point Options there.
b) Have PICSA Options as the only button and the Down arrow on this button has PICSA OPtions, Plot Options, Line Options, Point Options.

### PR DESCRIPTION
**Pull request by derekagorhom**
Fixes #7498 

Making changes in PICSA rainfall Graph, I went ahead with the first option, i.e 
Keep the PICSA Options as it is. So the 2 buttons remain. Also put the Down arrow on the Plot Options, and include Plot Options Line Options, Point Options there.

@N-thony @rdstern this PR is ready for review.

### CODE REVIEW

**Comment by lloyddewit** [comment]
@Ivanluv Please could you peer review? thanks

**Review by rdstern — CHANGES_REQUESTED** [review]
Good start.
a) The main Plot options doesn't work now.  When using the main button it should be the same as the down arrow and Plot Options.
b) The Point options, gives line Options and not points.

**Comment by derekagorhom** [comment]
Hello @rdstern, I have made the changes requested could you re-test again.

**Review by rdstern — APPROVED** [review]
looks good now

**Review by Ivanluv — DISMISSED** [review]

File: instat/dlgPICSARainfall.vb [hunk 1]
```diff
@@ -902,4 +892,126 @@ Public Class dlgPICSARainfall
...
+                          clsNewAnnotateFunction:=clsAnnotateFunction,
+                          clsNewGlobalAesFunction:=clsRaesFunction,
+                          ucrNewBaseSelector:=ucrSelectorPICSARainfall,
+                          bReset:=bResetSubdialog)
+        sdgPlots.ShowDialog()
+        bResetSubdialog = False
+        'AddRemoveGroupByAndHlines()
+    End Sub
+
+    Private Sub toolStripMenuItemPointOption_Click(sender As Object, e As EventArgs) Handles toolStripMenuItemPointOption.Click
```
Comment (lloyddewit) [inline instat/dlgPICSARainfall.vb:989]: what's the difference between this sub and `toolStripMenuItemLineOptions_Click()`?
Remove duplication?

**Review by lloyddewit — CHANGES_REQUESTED** [review]

**Comment by lloyddewit** [comment]
@derekagorhom Thank you for the changes. 
Please do not have lines of code that are more than 100 columns wide (it's bad for readability). I added line breaks. Please check that they are OK.
I also have one open comment above.
Thanks

File: instat/dlgPICSARainfall.vb (hunk 1 above)
Comment (derekagorhom) [reply instat/dlgPICSARainfall.vb:989]: Sorry these are not duplicates, one is for the line option and the other is for the Point option.

Comment (derekagorhom) [reply instat/dlgPICSARainfall.vb:989]: @lloyddewit the difference lies in the `sdgLayerOptions.SetupLayer`, the `LineOptions` has `clsNewGeomFunc:=clsGeomLine`
while the` PointOptions` has `clsNewGeomFunc:=clsPointsFunc`

**Comment by derekagorhom** [comment]
> @derekagorhom Thank you for the changes. Please do not have lines of code that are more than 100 columns wide (it's bad for readability). I added line breaks. Please check that they are OK. I also have one open comment above. If we can resolve this, then I think we can merge. Thanks

duly noted

**Comment by derekagorhom** [comment]
@lloyddewit i have made the changes requested, can you review again?

File: instat/dlgPICSARainfall.vb [hunk 2]
```diff
@@ -902,4 +863,100 @@ Public Class dlgPICSARainfall
             clsPipeOperator.RemoveAssignTo()
         End If
     End Sub
+
+
```
Comment (lloyddewit) [inline instat/dlgPICSARainfall.vb:867]:
```suggestion
```

File: instat/dlgPICSARainfall.vb [hunk 3]
```diff
@@ -902,4 +863,100 @@ Public Class dlgPICSARainfall
...
+                'Still might be in the case of bSingleVariable with mapping y="".
+                If clsParam.strArgumentValue = (Chr(34) & Chr(34)) Then
+                    ucrVariablesAsFactorForPicsa.Clear()
+                Else ucrVariablesAsFactorForPicsa.Add(clsParam.strArgumentValue)
+                End If
+            ElseIf clsParam.strArgumentName = "colour" Then
+                ucrReceiverColourBy.Add(clsParam.strArgumentValue)
+            End If
+        Next
+        TestOkEnabled()
```
Comment (lloyddewit) [inline instat/dlgPICSARainfall.vb:960]:
```suggestion
     openSdgLayerOptions(clsPointsFunc)
```

File: instat/dlgPICSARainfall.vb (hunk 1 above)
Comment (lloyddewit) [reply instat/dlgPICSARainfall.vb:934]: Can we call `openSdgLayerOptions()` here also? 
(see comment below)

**Review by lloyddewit — CHANGES_REQUESTED** [review]

**Comment by derekagorhom** [comment]
@lloyddewit i tried using `openSdgLayerOptions()` for the Point (I changed the `clsGeomLine` to `clsPointsFunc` i ended up with lots of errors and that's where the problem with the commits started

**Review by lloyddewit — DISMISSED** [review]
@rdstern If you can test/approve again, then we can merge, thanks

**Review by rdstern — CHANGES_REQUESTED** [review]
It is a detail, but I wonder if something can be done about this "feature".  I tried the Line option and found this:
image

The x was included, but not the y.  When I returned to the main dialogue, the y was now unset.  It had been set before, with the variable sumrain:
image

I tried also with the Dodoma data from the library, and the problem is the same.

**Comment by derekagorhom** [comment]
@rdstern i have made some changes to the dialogue and the sub dialogues. could you Re-test

**Review by rdstern — APPROVED** [review]
Great.  Well done.

**Review by lloyddewit — APPROVED** [review]

**Pull request merged**